conninfo/src/conninfo-v1.0-wd01.md

```diff
@@ -0,0 +1,238 @@
+
+-------
+
+# 2 Overview
+
+This specification recommends a standard way for an AMQP container to reveal information about
+itself and its operating environment to its partner.   This allows a container to expose information
+that would be useful for problem diagnosis, such as process identifiers and software version
+information.
+
+This specification is equally appropriate for AMQP container fulfilling the role of clients or
+servers, or containers supporting a peer-to-peer scenario.
+
+-------
+
+# 3 Definitions
+
+An AMQP host that implements this specification MUST provide the following entry in the properties
+field of the AMQP `open` performative [[AMQP-v1.0](#AMQP-v1.0)]. The key of the entry MUST be the symbol
+`connection-info`. The value of the entry MUST be of type `fields`.
+
+The following key values have defined meaning. An implementation MAY choose to omit a key from the map.
+An implementation MAY also include keys of its own. If an implementation receives a key from its partner
+that it does not recognise, it SHOULD ignore it.
+
+# 3.1 connection-info
+
+`<type name="connection-info" class="restricted" source="fields"/>`
+
+The content of `connection-info` allows the container to provide information about itself and its operating
+environment to its partner.
+
+**Table 1-1. connection-info**
+
+| Name | Description |
+| :--- | :--- |
+| `process-identifier` | The value of the process-identifier MUST be an AMQP string. The purpose of the value is to identify the process running this AMQP host. It is RECOMMENDED that this is an operating system PID. |
+| `network-host` | The value of network-host MUST be an AMQP string. The purpose of the value is to identify the host running the AMQP container in a way that assists those fault finding within a system. This value MAY be a hostname, fully qualified hostname or IP address. The value MAY be network resolvable by the AMQP peer but the peer MUST NOT rely on this. |
+| `amqp-product` | The value of amqp-product MUST be an AMQP string and is a `product-string`. It provides details of the software product(s) that provide the AMQP implementation. It MAY identify multiple products. By convention, the products are listed in order of their significance for identifying the product. |
+| `product` | The value of product MUST be an AMQP string and is a product-string. It provides details of the software product(s) that is the end-user application. It MAY identify multiple products. By convention, the products are listed in order of their significance for identifying the application. |
+| `platform` | The value of platform MUST be an AMQP string and is a product-string. It exists to expose details of a product’s runtime platform. This typically identify the operating system, Java/.NET runtimes etc. It MAY identify multiple products. By convention, the products are listed in order of their significance. |
```

Comment:
I think you can omit the "an AMQP string and is" in all three properties. The product-string definition does that job.

Reply:
If we have three buckets for identifiers, we need to have clearer lines between those buckets. What if the .NET Framework had a native implementation AMQP in the box and used AMQP for its own diagnostics telemetry; where would that go?

Reply:
I'm sort of OK with these classifications, but I admit they are a bit blurry.  I think "platform" is something that is only known at run time, and thus distinct from the other two buckets.  AMQP Product vs. product is probably a bit blurrier, but at this point I don't imagine many people will be building AMQP libraries from scratch... so I think this is sort of OK... It will be baked into that library code and not modifiable at runtime, or even at the point at which an application is built.

Reply:
@clemensv I think if a product baked AMQP right into itself and didn't rely on a AMQP library for support (which I think is your example), then I would omit the `amqp-product` from the map altogether.  

@rgodfrey for `platform`, I think the AMQP library would derive this using information such as `java.runtime.version`, `os.name`, `os.version`, `os.arch` in the Java world, or uname(2) in POSIX.

I used the words "end-user application" for `product` which is misleading.  The intent is that a server side product which does not necessarily have a end-user populates this too.   I'll refine these definitions.

Reply:
(Robbie Gemmell): `amqp-product` should be better as `amqp-component` and that `product` ought to be `application`.  We mentioned that the property name `product` was already in use by several AMQP implementations to convey the AMQP product information.  The desire is to avoid confusion with that.

Reply:
I've actioned Robbie's comments.